cmd/attest.go

```diff
@@ -0,0 +1,267 @@
+	}
+}
+
+func attestExec(ctx context.Context, _ *cobra.Command, args []string) error {
+	// can only be an image for attestation or OCI DIR
+	userInput := args[0]
+	err := validateScheme(userInput)
+	if err != nil {
+		return err
+	}
+
+	ko := sign.KeyOpts{
+		KeyRef:   keyPath,
+		PassFunc: passFunc,
+	}
+
+	sv, err := sign.SignerFromKeyOpts(ctx, "", ko)
+	if err != nil {
+		return err
+	}
+	defer sv.Close()
+
+	return eventLoop(
+		attestationExecWorker(userInput, sv),
+		setupSignals(),
+		eventSubscription,
+		stereoscope.Cleanup,
+		ui.Select(isVerbose(), appConfig.Quiet)...,
+	)
+}
+
+func attestationExecWorker(userInput string, sv *sign.SignerVerifier) <-chan error {
+	errs := make(chan error)
+	go func() {
+		defer close(errs)
+		if len(attestationOutput) > 1 {
+			errs <- fmt.Errorf("can not generate attestation for more than one output")
+			return
+		}
+
+		output := format.ParseOption(attestationOutput[0])
+		if output == format.UnknownFormatOption {
+			errs <- fmt.Errorf("can not use %v as attestation format. Try: %v", output, format.AllOptions)
+			return
+		}
+
+		s, src, err := generateSBOM(userInput, errs)
+		if err != nil {
+			errs <- err
+			return
+		}
+
+		bytes, err := syft.Encode(*s, output)
+		if err != nil {
+			errs <- err
+			return
+		}
+
+		err = generateAttestation(bytes, src, sv, output)
+		if err != nil {
+			errs <- err
+			return
+		}
+	}()
+	return errs
+}
+
+func assertPredicateType(output format.Option) string {
+	switch output {
+	case format.SPDXTagValueOption, format.SPDXJSONOption:
+		return in_toto.PredicateSPDX
+	default:
+		return in_toto.StatementInTotoV01
+	}
+}
+
+func generateAttestation(predicate []byte, src *source.Source, sv *sign.SignerVerifier, output format.Option) error {
+	predicateType := assertPredicateType(output)
+
+	h, _ := v1.NewHash(src.Image.Metadata.ManifestDigest)
+
+	wrapped := dsse.WrapSigner(sv, "application/vnd.in-toto+json")
+
+	sh, err := attestation.GenerateStatement(attestation.GenerateOpts{
+		Predicate: bytes.NewBuffer(predicate),
+		Type:      predicateType,
+		Digest:    h.Hex,
+	})
+	if err != nil {
+		return err
+	}
+
+	payload, err := json.Marshal(sh)
+	if err != nil {
+		return err
+	}
+
+	signedPayload, err := wrapped.SignMessage(bytes.NewReader(payload), signatureoptions.WithContext(context.Background()))
+	if err != nil {
+		return errors.Wrap(err, "signing")
+	}
+
+	bus.Publish(partybus.Event{
+		Type: event.Exit,
+		Value: func() error {
+			_, err := os.Stdout.Write(signedPayload)
+			return err
+		},
+	})
+
+	return nil
+}
+
+func init() {
+	setAttestFlags(attestCmd.Flags())
+	rootCmd.AddCommand(attestCmd)
+}
+
+func setAttestFlags(flags *pflag.FlagSet) {
+	// Key options
+	flags.StringVarP(&keyPath, "key", "", "cosign.key",
+		"private key to use to sign attestation",
+	)
+
+	flags.StringArrayVarP(&attestationOutput,
+		"output", "o", []string{string(format.JSONOption)},
+		fmt.Sprintf("attestation output format, options=%v", format.AllOptions),
```

Comment:
nit: the phrasing makes it seem like this controls the attestation format instead of whats within the attestation, maybe something like
```suggestion
		fmt.Sprintf("SBOM output format, options=%v", format.AllOptions),
```

Reply:
updated phrasing